### PR TITLE
Feat/411 2 chronik skeleton and get balance

### PR DESCRIPTION
**Pull request by vorg-code**
Steps to test:

1. This PR should be reviewed after #417 since it requires those packages (they are included here as well so they could be reviewed simultaneously, however if there are any breaking changes there, things could get complex to merge afterwards)
2. open in you browser and see balance (example address in this url, feel free to change it): http://localhost:3000/api/address/balance/ecash:prezk74rzl4uw22wm6agcteuwztdvp5a9umns6k0d3
3. Change in constants/index.ts the ecash choice to chronik:
`export const BLOCKCHAIN_CLIENTS_CHOSEN: KeyValueT<BLOCKCHAIN_CLIENTS_OPTIONS> = {
  ecash: 'chronik',
  bitcoincash: 'grpc'
}`
4. open in you browser and see balance (now from chronik client): http://localhost:3000/api/address/balance/ecash:prezk74rzl4uw22wm6agcteuwztdvp5a9umns6k0d3

Other API endpoints will not work with chronik as they are not implemented yet, this will be addressed in future PRs.

### CODE REVIEW

**Review by Klakurka — APPROVED** [review]

**Comment by Klakurka** [comment]
@chedieck can do a code review on this before we merge it.

**Review by chedieck — COMMENTED** [review]
Left some code remarks.

Also, I see that `tests/unittests/grpcService.test.ts` was changed to import `blockchainService` instead of `grpcService`, even though the test file name wasn't changed. 

Concerning the unittests, I think we should test `grpcService` and `chronikService` seperately, or just test `blockchainService` but using both clients.

File: constants/index.ts [hunk 1]
```diff
@@ -96,3 +98,9 @@ export const NETWORK_TICKERS: KeyValueT<string> = {
 
 export const NETWORK_IDS: KeyValueT<number> = { XEC: 1, BCH: 2 }
 export const QUOTE_IDS: KeyValueT<number> = { USD: 1, CAD: 2 }
+
+export type BLOCKCHAIN_CLIENTS_OPTIONS = 'grpc' | 'chronik'
```
[the remark this answers is not in the document]
Comment (vorg-code) [reply constants/index.ts:102]: done

File: constants/index.ts [hunk 2]
```diff
@@ -96,3 +98,9 @@ export const NETWORK_TICKERS: KeyValueT<string> = {
...
+export const BLOCKCHAIN_CLIENTS_CHOSEN: KeyValueT<BLOCKCHAIN_CLIENTS_OPTIONS> = {
```
[the remark this answers is not in the document]
Comment (vorg-code) [reply constants/index.ts:103]: done

File: services/blockchainService.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  getTransactionDetails: (hash: string, networkSlug: string) => Promise<GetTransactionResponse.AsObject>
+  subscribeTransactions: (
+    addresses: string[],
+    onTransactionNotification: (txn: Transaction.AsObject) => any,
+    onMempoolTransactionNotification: (txn: Transaction.AsObject) => any,
+    networkSlug: string
+  ) => Promise<void>
+}
+
+function getBlockchainClient (network: string): BlockchainClient {
```
[the remark this answers is not in the document]
Comment (vorg-code) [reply services/blockchainService.ts:38]: done

File: utils/index.ts [hunk 4]
```diff
@@ -121,3 +121,14 @@ export async function fileExists (fsModule: any, filePath: string): Promise<bool
 export function isEmpty (value: string): boolean {
   return value === '' || value === null || value === undefined
 }
+
+export function getObjectForAddress<T> (addressString: string, objects: KeyValueT<T>): T {
```
[the remark this answers is not in the document]
Comment (vorg-code) [reply utils/index.ts:125]: ok, done

File: services/chronikService.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    let totalSatoshis = 0
+    // we should always get only one element since we have only one hash above
+    if (utxos.length > 0) {
+      for (const u of utxos[0].utxos) { totalSatoshis += parseInt(u.value) }
+    }
+
+    return totalSatoshis
+  }
+
+  private toHash160 (address: string): {type: ScriptType, hash160: string} {
```
[the remark this answers is not in the document]
Comment (vorg-code) [reply services/chronikService.ts:51]: Can I do it in the next PR on this chain? these 2 methods aren't even in this file anymore. If I do this in this PR, there will be merge conflict issues later on.

**Comment by vorg-code** [comment]
@chedieck your code remarks have been addressed/replied.

About the tests, we will discontinue grpc after these streams of PR go live, so I will only create tests for the chronik implementation. There is this PR here ready, the next one also is ready #428 and the next one I have almost finished coding. Can I address the testing of the 3 PRs in this next one that is almost done? Or maybe even to an additional PR at the end with all the chronik tests. This way we can minimize merging conflicts.

**Review by chedieck — APPROVED** [review]
LGTM, also good idea to do the refactoring of the tests in another separate PR

**Pull request merged**